### PR TITLE
Add Prometheus Exports to Mongodb Replicaset

### PR DESCRIPTION
**Pull request by ssalaues**
**What this PR does / why we need it**: 
1. Adds option for prometheus compatible metric exports in all pods under the Mongodb-replicaset chart
2. Fixes some broken TLS chart functionality and allows the metrics exporter to use TLS options

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
First PR into kubernetes/charts 😁

### CODE REVIEW

**Comment by ssalaues** [comment]
Just signed the CLA with my github! 😄

Also I added dockerfiles for the container but not necessary to the functionality. If it's not wanted, I can drop those files which would reduce the size of this PR.

File: stable/mongodb-replicaset/Chart.yaml [hunk 1]
```diff
@@ -1,12 +1,13 @@
 name: mongodb-replicaset
 home: https://github.com/mongodb/mongo
-version: 3.4.0
+version: 3.4.1
```
Comment (prydonius) [inline stable/mongodb-replicaset/Chart.yaml]: this should be bumped to `3.5.0` as this is a new feature

**Review by prydonius — COMMENTED** [review]
thanks! lgtm in general, the minor version needs to be bumped as this is a new feature (we follow semver).

/ok-to-test

**Comment by ssalaues** [comment]
Thanks for the feedback! I bumped the version to 3.5 and fixed the conflicts.

**Comment by unguiculus** [comment]
@prydonius @ssalaues I'm inclined to revert this PR.

* Always adds annotation for Prometheus. You don't need them if you use prometheus-operator.
* The Makefile has stuff specific to Google Container Registry in it but the image is on Dockerhub. Why?
* Doesn't follow best practices for specifying images.
* Most importantly: Uses the admin user for the exporter instead of a user with the `clusterMonitor` role. This is bad practice and a security risk. We should not have such practices in the chart.

**Comment by prydonius** [comment]
@unguiculus sure, I think we should go ahead and revert this. Once reverted, @ssalaues can re-create a PR to apply the changes and we can try to solve some of the issues you mentioned.

**Comment by ssalaues** [comment]
@unguiculus I'll make the requested changes and re create the PR. Thanks!

**Comment by unguiculus** [comment]
@ssalaues Thanks. I'll be happy to assist. You can find my on Kubernetes Slack.

**Pull request merged**